src/file/FileDialog.tsx

```diff
@@ -215,6 +218,74 @@ const OpenLocalFile: React.FC<SourceTabProps> = ({ onDismiss }) => {
     );
 };
 
+const ImportFromString: React.FC<SourceTabProps> = ({ onDismiss }) => {
+    const loadScene = useLoadScene();
+    const setSavedState = useSetSavedState();
+    const isDirty = useIsDirty();
+    const theme = useTheme();
+    const [data, setData] = useState<string | undefined>('');
+    const [error, setError] = useState<string | undefined>('');
+
+    const importCallback = useCallback(async () => {
+        if (!data) {
+            return;
+        }
+
+        if (isDirty) {
+            if (!(await confirmUnsavedChanges(theme))) {
+                return;
+            }
+        }
+
+        let scene;
+        try {
+            scene = await textToScene(data);
+        } catch (ex) {
+            console.log(ex);
+            setError('Invalid Import String');
```

Comment:
Provided the error messages are usually reasonably short and understandable, I think including the error would be useful since most people won't think to check the console when reporting an issue. See also my comment below about it not being very clear what an "import string" is. Maybe something like this?

```suggestion
            setError(`Failed to import: ${ex}`);
```

Reply:
The error messages I was able to produce were:
* `DOMException: String contains an invalid character` when I enter invalid base64 characters.
* `incorrect header check` when I enter valid base64, but the data cannot be decompressed.
* Various JSON parser exceptions:
* `SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data`
* `SyntaxError: JSON.parse: expected property name or '}' at line 1 column 2 of the JSON data`

If I was a non technical end user and I saw any of these errors, I would assume that there was a bug in the app, rather than a mistake on my end. Also, when reporting issues users should include the plan code in the report, which is everything a developer needs to reproduce the problem.

Because of this, I believe that including the error message in the UI would not be beneficial. It'll still be available in the developer console.